src/Symfony/Bundle/SecurityBundle/Security.php

```diff
@@ -59,7 +59,9 @@ public function getFirewallConfig(Request $request): ?FirewallConfig
      */
     public function login(UserInterface $user, string $authenticatorName = null, string $firewallName = null): void
     {
-        $request = $this->container->get('request_stack')->getCurrentRequest();
+        if (!$request = $this->container->get('request_stack')->getCurrentRequest()) {
```

Comment:
Let's be explicit that we expect null here. We should also do it in two lines so it is easier to read.

```suggestion
        $request = $this->container->get('request_stack')->getCurrentRequest();
        if (null === $request) {
```

Reply:
Hi @Nyholm , I appreciate your help.
 - I fixed the CS as you suggested.
 - I added a test for exceptions in `login()` and `logout()`

I realized that [**login** uses **getCurrentRequest**](https://github.com/symfony/symfony/blob/7.1/src/Symfony/Bundle/SecurityBundle/Security.php#L86), and [**logout** uses **getMainRequest**](https://github.com/symfony/symfony/blob/7.1/src/Symfony/Bundle/SecurityBundle/Security.php#L118). Do you think this is inconstancy is on purpose ?

Reply:
Lovely!

> I realized that [login uses getCurrentRequest](https://github.com/symfony/symfony/blob/7.1/src/Symfony/Bundle/SecurityBundle/Security.php#L86), and [logout uses getMainRequest](https://github.com/symfony/symfony/blob/7.1/src/Symfony/Bundle/SecurityBundle/Security.php#L118). Do you think this inconstancy is on purpose ?

Yes, that is on purpose. I guess "logout" should logout from "all". But "login" needs the context from the current request and not the main request.

Reply:
ok, makes sense now :)